test/test_losses.py

```diff
@@ -340,3 +340,53 @@ def test_tv_on_invalid_dims(self, input):
     def test_tv_on_invalid_types(self, input):
         with pytest.raises(TypeError) as ex_info:
             kornia.losses.total_variation(input)
+
+
+class TestPSNR:
+    def test_smoke(self):
+        signal = torch.rand(2, 3, 3, 2)
+        approximation = torch.rand(2, 3, 3, 2)
+
+        criterion = kornia.losses.PSNR(1.0)
+        loss = criterion(signal, approximation)
+
+        assert loss.shape == tuple()
+
+    def test_same_signal(self):
+        signal = torch.rand(2, 3, 3, 2)
+        approximation = signal
+
+        criterion = kornia.losses.PSNR(1.0)
+        loss = criterion(signal, approximation)
+
+        assert pytest.approx(loss.item(), float('inf'))
+
+    def test_type(self):
+        # Expecting an exception
+        # since we pass integers instead of torch tensors
+        criterion = kornia.losses.PSNR(1.0)
+        with pytest.raises(Exception) as e:
+            criterion(1, 2)
+
+    def test_shape(self):
+        # Expecting an exception
+        # since we pass tensors of different shapes
+        criterion = kornia.losses.PSNR(1.0)
+        with pytest.raises(Exception) as e:
+            criterion(torch.rand(2, 3, 3, 2), torch.rand(2, 3, 3))
+
+    def test_simple(self):
+        assert pytest.approx(kornia.losses.psnr(torch.ones(1), 1.2 * torch.ones(1), 2).item(), 20.0)
+
+    @pytest.mark.skip(reason="TODO: implement me")
+    def test_jit(self):
+        pass
+
+    def test_gradcheck(self):
+        signal = torch.rand(2, 3, 3, 2).double()
```

Comment:
`tensor_to_gradcheck_var` already does that for you

Reply:
Sorry, I do not know what it meant here. I was kind of following an existing test here: https://github.com/kornia/kornia/blob/master/test/test_losses.py#L99

Reply:
`tensor_to_gradcheck_var` cast to double precision. See: https://github.com/kornia/kornia/blob/master/kornia/testing/__init__.py#L36